internal/cmd/environment/command_create.go

```diff
@@ -41,7 +41,8 @@ func (c *command) create(cmd *cobra.Command, args []string) error {
 	}
 
 	c.Context.AddEnvironment(environment.GetId())
+	c.Config.ValidEnvNamesToIds[environment.GetDisplayName()] = environment.GetId()
+	c.Config.ValidEnvIdsToNames[environment.GetId()] = environment.GetDisplayName()
```

Comment:
What's the reasoning for storing this in the configuration file? 🤔

Reply:
The environment variable in the dynamic context is just a string, so there's no way to verify that it's actually a valid name corresponding to an ID without calling the list API (which we want to avoid calling if possible) if it is passed into an arbitrary command as a flag. Storing a mapping of environment names -> ids + vice versa in the config makes more sense here since we also want the mapping to persistent across multiple commands after the first time a user tries using the `--environment` flag (doesn't make sense to populate the map before then either because the current environment is assumed in the config).

Reply:
I just realized but I probably want to do the same with `--cluster` too if you think it's worth it to basically "cache" these two mappings in the config to avoid extra list API calls. My concern is that the user's config might become very cluttered by these mappings. But if we choose not to maintain the mappings in some persistent way, every command that is run with a `--cluster/--environment` flag where the value's prefix is not the correct one would result in the list API being called in order to check if the value is a valid resource name. How should we value the time vs space tradeoff here? I think I would prefer to maintain the mapping some way (if we can avoid the config that's probably the best).

Reply:
> The environment variable in the dynamic context is just a string

What does this mean?

> Storing a mapping of environment names -> ids + vice versa in the config makes more sense here

We've gotten in trouble for doing this in the past. What if you store an ID -> name mapping in the configuration file, and then change the name of the resource on the website?

Reply:
> > The environment variable in the dynamic context is just a string
> 
> What does this mean?
> 
The Environment field in the context struct is just a string, not the orgV2Environment struct that is returned by the API. Maybe we should change this?

Reply:
> We've gotten in trouble for doing this in the past. What if you store an ID -> name mapping in the configuration file, and then change the name of the resource on the website?

Yeah the config mapping would have no idea about UI changes to resources. I think one way we can avoid this is to retry a command when it fails when using `--cluster` or `--environment` and only then make the API call.

Reply:
> The Environment field in the context struct is just a string, not the orgV2Environment struct that is returned by the API.

That's intentional. The idea is that's the only field in the struct that's not subject to change (i.e. by interacting with the UI)

Reply:
> This would worst case result in 2 extra list API calls on top of the command being run if both the cluster and environment are passed in as names rather than IDs.

Yeah, I think this is the best we can do. Fortunately, both of these calls will return a few pages of results as the worst case. What we want to watch out for is any resources that a user might potentially have hundreds or thousands of, in which case we might need to make 100s of paginated API calls.

Reply:
For flags it looks like there's only `identity-provider`, `ksql-cluster`, `service-account` besides the two mentioned above.